### PR TITLE
Change type of Question#validation_rules to JSON

### PR DESCRIPTION
**Pull request by leonhooijer**
I only needed to change te column type in the migration. We could think about adding a rake task for migrating existing applications.

### CODE REVIEW

**Comment by leonhooijer** [comment]
This implements #48.

**Comment by leonhooijer** [comment]
There seems to be an inconsistency between the local specs and the Travis CI builds. Locally the specs pass with multiple configurations, but I can't get the Travis CI builds to pass. I will take a look at this again at a later moment (with a clear mind).

File: db/migrate/20130502195310_create_rapidfire_questions.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def change
       t.string  :question_text
       t.integer :position
       t.text :answer_options
-      t.text :validation_rules
+      t.column :validation_rules, :jsonb, null: false, default: '{}'
```
Comment (iffyuva) [inline db/migrate/20130502195310_create_rapidfire_questions.rb]: @leonhooijer does this work with sqlite3, mysql?

**Comment by leonhooijer** [comment]
I'm getting close. 

For the JSON(B) column type to work I used PostgreSQL instead of SQLite. SQLite does have a [JSON extension](https://www.sqlite.org/json1.html), but I don't use SQLite enough know how to automate the setup in the project. If anyone has any suggestions please let me know.

It turns out [Travis CI uses PostgreSQL 9.1 by default](https://docs.travis-ci.com/user/database-setup/#Using-a-different-PostgreSQL-Version). For the JSON(B) data type to work we need at least version 9.4. I tried 9.5 but is is not supported out of the box right now, so I decided to stick with 9.4.

Furthermore I needed to use the `t.column :validation_rules, :jsonb` since `t.jsonb` is not supported in Rails versions older then 4.2.

**Comment by leonhooijer** [comment]
So the Travis CI build finally passes, but I guess it is a good idea to do a thorough code review.

**Comment by iffyuva** [comment]
hi @leonhooijer thanks for this work. btw, we have to support mysql also, because couple of our clients use this gem. so, dropping support for mysql is not a good idea. lets think about ways to support both pg and mysql. i will drop in suggestions that i have here :)

**Comment by leonhooijer** [comment]
So, after experiencing some real world situations I believe we should go for the JSON column type instead of the JSONB column type. The JSON column type is supported by more database types and also by older PostgreSQL databases. It is less efficient then the JSONB column type, but I believe the extra compatability is worth it.

I will change this soon.

**Comment by iffyuva** [comment]
@leonhooijer any update on this? Also, can you make sure that this code works with mysql. Its not a good idea if we break people who use mysql. Thanks.

**Comment by leonhooijer** [comment]
Thanks for the reminder @iffyuva 😉.

What I did today was just a rebase on master. I tested with MySQL a bit locally and it seems to work. Since the project only has tests for SQLite it would be quite a bit of work to also test this with MySQL and PostgreSQL, because we need to test combinations of Rails versions with different types of databases (and maybe even different database versions). But it seems like a good idea to me.

This solution works with MySQL 5.7 and above and PostgreSQL 9.1 and above, since the JSON column type was introduced in these versions.

**Pull request closed without merging**